plugins/modules/monitoring/nagios.py

```diff
@@ -308,95 +304,44 @@ def main():
 
     module = AnsibleModule(
         argument_spec=dict(
-            action=dict(required=True, choices=ACTION_CHOICES),
-            author=dict(default='Ansible'),
-            comment=dict(default='Scheduling downtime'),
-            host=dict(required=False, default=None),
-            servicegroup=dict(required=False, default=None),
-            start=dict(required=False, default=None),
-            minutes=dict(default=30, type='int'),
-            cmdfile=dict(default=which_cmdfile()),
-            services=dict(default=None, aliases=['service']),
-            command=dict(required=False, default=None),
-        )
+            action=dict(type='str', required=True, choices=ACTION_CHOICES),
+            author=dict(type='str', default='Ansible'),
+            comment=dict(type='str', default='Scheduling downtime'),
+            host=dict(type='str'),
+            servicegroup=dict(type='str'),
+            start=dict(type='str'),
+            minutes=dict(type='int', default=30),
+            cmdfile=dict(type='str', default=which_cmdfile()),
+            services=dict(type='str', aliases=['service']),
+            command=dict(type='str'),
+        ),
+        required_if=[
+            ('action', 'downtime', ['host', 'services']),
+            ('action', 'delete_downtime', ['host', 'services']),
+            ('action', 'silence', ['host']),
+            ('action', 'unsilence', ['host']),
+            ('action', 'enable_alerts', ['host', 'services']),
+            ('action', 'disable_alerts', ['host', 'services']),
+            ('action', 'silence_nagios', ['services']),
+            ('action', 'unsilence_nagios', ['services']),
+            ('action', 'command', ['services', 'command']),
```

Comment:
Also here, `services` does not seem to be required:
```suggestion
            ('action', 'command', ['command']),
```